svelte-web-frontend/src/lib/data/dances-store.ts

```diff
@@ -129,7 +129,25 @@ export class PoseReferenceData<T extends Pose2DPixelLandmarks | Pose3DLandmarkFr
         // const frameIndex = this.frameIndices[searchIndex];
         const pose = this.poses[searchIndex];
 
-        return pose;
+        return pose ?? null;
+    }
+
+    /**
+     * Get the reference pose information for an array of dance timestamps.
+     * @param frameTimes Array of dance timestamps for which to get the pose information
+     * @returns Array of pose information for the frames at the given timestamps, or an empty array if no pose information is available for any frame
+     */
+    get2DLandmarks(frameTimes: number[]): Pose2DPixelLandmarks[] {
+        const poses: Pose2DPixelLandmarks[] = [];
+
+        for (const timestamp of frameTimes) {
```

Comment:
this doesn't match the behavior you describe in the comment above. In the coment, you say you'll return an empty array for frame where no pose information is avaiable. This would ensure the characteristic that the return `poses` array will always be the same length as the input `frameTimes` array.

However, in your implementation, you don't push empty arrays when the pose is null - you simply omit the frame entirely. This means that the output `poses` array won't be the same length as `frameTimes` if there are any frames that miss pose information, which will also means that the output array won't be aligned with the input array. Suppose frame 39 and 40 are missing pose information. Then, poses[0...38] and frameTimes[0...38] would be aligned. But then frameTimes[39...40] would exist, but poses[39...40] would actual be the poses that _should_ correspond to frames 41 and 42. Furthermore, every every frameTime[41...n] would be 2 frames misaligned with the poses array, and this would only get worse in the presence of additional missing frames.

This isn't something you need to fix before merging this PR, as the refactoring I've done makes this irrelevant. However, any metric we implement needs to be able to handle missing frames, and it shouldn't do so by discarding the contents, as that can skew the results. We can talk more about this in our next meeting :).

Reply:
Yes, I see the problem. I tried to modify this method but the 2D / 3D change is giving me a lot of bugs. I will change that after the merge.